fe/fe-core/src/main/java/com/starrocks/connector/delta/ScalarOperationToDeltaLakeExpr.java

```diff
@@ -0,0 +1,384 @@
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.IsNullPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.LikePredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperatorVisitor;
+import com.starrocks.sql.optimizer.operator.scalar.SubfieldOperator;
+import io.delta.kernel.expressions.And;
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.expressions.Or;
+import io.delta.kernel.expressions.Predicate;
+import io.delta.kernel.types.StructField;
+import io.delta.kernel.types.StructType;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.delta.kernel.expressions.AlwaysTrue.ALWAYS_TRUE;
+
+public class ScalarOperationToDeltaLakeExpr {
+    public Predicate convert(List<ScalarOperator> operators, DeltaLakeContext context) {
+        DeltaLakeExprVisitor visitor = new DeltaLakeExprVisitor();
+        List<Predicate> predicates = Lists.newArrayList();
+
+        for (ScalarOperator operator : operators) {
+            Predicate predicate = operator.accept(visitor, context);
+            if (predicate != null) {
+                predicates.add(predicate);
+            }
+        }
+
+        Optional<Predicate> result = predicates.stream().reduce(And::new);
+        return result.orElse(ALWAYS_TRUE);
+    }
+
+    public static class DeltaLakeContext {
+        private final io.delta.kernel.types.StructType schema;
+        private final Set<String> partitionColumns;
+
+        public DeltaLakeContext(io.delta.kernel.types.StructType schema, Set<String> partitionColumns) {
+            this.schema = schema;
+            this.partitionColumns = partitionColumns;
+        }
+
+        boolean isPartitionColumn(String column) {
+            return partitionColumns.contains(column);
+        }
+
+        Column getColumn(String columnName) {
+            if (schema == null) {
+                return null;
+            }
+            int index = schema.indexOf(columnName);
+            if (index < 0) {
+                return null;
+            }
+            return schema.column(index);
+        }
+
+        public io.delta.kernel.types.StructType getSchema() {
+            return schema;
+        }
+    }
+
+    private static class DeltaLakeExprVisitor extends ScalarOperatorVisitor<Predicate, DeltaLakeContext> {
+        private static DeltaDataType getResultType(String columnName, DeltaLakeContext context) {
+            Preconditions.checkNotNull(context);
+            return getColumnType(columnName, context);
+        }
+
+        private static DeltaDataType getColumnType(String qualifiedName, DeltaLakeContext context) {
+            //TODO: nested type
```

Comment:
The expr generated by this transformation is only used for clipping. What was issued to BE was a different set. Mainly because this reduction based on Struct type has not been fully tested, and the existing DeltaLake implementation does not support it, so we will stay on the battlefield here first.